src/app/zones/views/CreateView.vue

```diff
@@ -349,7 +370,7 @@
         :title="t('zones.form.confirm_modal.title')"
         data-testid="confirm-exit-modal"
         @canceled="toggleConfirmModal"
-        @proceed="router.push({ name: 'zone-cp-list-view' })"
+        @proceed="route.replace({ name: 'zone-cp-list-view' })"
```

Comment:
**Question**: Why is this being changed? Doesn’t seem necessary.

We should use the standard route utilities as much as we can so we don’t accidentally build a whole custom router on top of vue-router.

Reply:
We've spoken about this lots of times before. We are replacing the current route here.

> We should use the standard route utilities as much as we can so we don’t accidentally build a whole custom router on top of vue-router.

We aren't building a whole new custom router on top of vue-router. Where do you think we are doing that? We did used to have completely re-implemented query parameter interactions, which I think you yourself added (?) but we've removed all that now purposefully to use vue-router instead of having our own implementation.

What we do have is a thin layer over `vue-router`s `router.push`. This means we can smooth over rough edges (should we find them) in a centralized place should we need to. This means we don't have to re-implement the same fixes over and over again throughout the application. This means that we don't get in a state where some fixes are applied in one place but not another. Something that we still have quite a lot of, but gradually less due to things approaches like this.

We have already used this thin layer over vue-router to fix the interactions between vue-router and pagination history, I would guess at some point in the future we will use it again.

Reply:
We so far haven’t used RouteView’s `route.replace` for something that we couldn’t have accomplished with an **equivalent** `router.push` call. The implementation of the abstraction is calling `router.push` directly without any changes:

```ts
const routeReplace = (...args: Parameters<typeof router['push']>) => {
  router.push(...args)
}
```

This means there isn’t actually a need for a fix or change of `router.push` behavior and we shouldn’t provide the abstraction in the first place.

Abstractions that actually abstract something away can be good. This here is a case of obfuscating a `router.push` call without there being a need for it.

Reply:
The important phrase is "so far".

We've spoken about `route.replace` (its not `router.push`) many many times before. I think the bigger thing is...

We've spoken about using the utilities that `RouterView` provides many many times before. We use them as it gives us centralized points to change/fix/amend things, should we need to, at an application global level.

This is what it is doing. This is the "feature" - being able to change/fix/amend things at an application level. We don't yet, but the "feature" is being able to easily.

Consequently, if we do use it to fix things, it means we don't have various fixes dotted all over the application that (try to) achieve the same thing with completely different code. Something I'm trying to reduce.

Please could we stop discussing not using the utilities that `RouteView` provides. Unless there is an actual technical reason not to, we should use them in all Route views.

Reply:
I think I can resolve this under "when a RouteView utility is available we should use it", let me know if there is still anything unresolved here. Resolving.